src/Arg.roc

```diff
@@ -385,14 +386,13 @@ andMap = \@Parser parser, @Parser mapper ->
 program = \parser, { name, help ? "" } ->
     @NamedParser { name, help, parser }
 
-# TODO panics in alias analysis when this annotation is included
-# parse : NamedParser a, List Str -> Result a (ParseError*)
 ## Parses a list of command-line arguments with the given parser. The list of
 ## arguments is expected to contain the name of the program in the first
 ## position.
 ##
 ## If the arguments do not conform with what is expected by the parser, the
 ## first error seen will be returned.
+parse : NamedParser a, List Str -> Result a (ParseError [])
```

Comment:
Sweet that this works now! 😃

Reply:
Actually I think @ayazhafiz's changes to tag unions mean we can now get rid of the type variable in `ParseError`. It should be possible to return a tag union with no type variable on it, and it should still be able to compose with other error unions!

Reply:
Changed, and it works 🎉